src/compiler/checker.ts

```diff
@@ -5097,6 +5097,13 @@ namespace ts {
                 let targetSig = targetSignatures[0];
 
                 if (sourceSig && targetSig) {
+                    if (kind === SignatureKind.Construct) { 
+                        result &= signatureVisibilityRelatedTo(sourceSig, targetSig);
+                        if (result !== Ternary.True) {

```

Comment:
I thought I should follow the same structure as [here](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/checker.ts#L5087).

It checks to see if its not assignable (resulted from ```signatureVisibilityRelatedTo```) and returns if not.